code/ui/generalhud/menu/Menu.cs

```diff
@@ -52,27 +53,86 @@ public void OpenSettings(PanelContent menuContent)
-                panelContent.Add.Label("Test");
-                panelContent.Add.Label("Test");
-                panelContent.Add.Label("Test");
-                panelContent.Add.Label("Test");
+                Panel buttonsWrapperPanel = panelContent.Add.Panel("wrapper");
+
+                buttonsWrapperPanel.Add.Button("Save as", "fileselectionbutton", () =>
+                {
+                    FileSelection fileSelection = FindRootPanel().Add.FileSelection();
+                    fileSelection.DefaultSelectionPath = "/settings/";
+
+                    fileSelection.OnAgree = () =>
+                    {
+                        string fileName = fileSelection.FileNameEntry.Text;
+
+                        if (string.IsNullOrEmpty(fileName))
+                        {
+                            return;
+                        }
+
+                        Settings.SettingFunctions.SaveSettings(fileName.Split('/')[^1].Split('.')[0]);
+
+                        fileSelection.Close();
+                    };
+
+                    fileSelection.EnableFileNameEntry();
+                    fileSelection.Display();
+                });
+
+                buttonsWrapperPanel.Add.Button("Load from", "fileselectionbutton", () =>
+                {
+                    FileSelection fileSelection = FindRootPanel().Add.FileSelection();
+                    fileSelection.DefaultSelectionPath = "/settings/";
+
+                    fileSelection.OnAgree = () =>
+                    {
+                        string fileName = fileSelection.SelectedEntry.FileNameLabel.Text;
+
+                        if (string.IsNullOrEmpty(fileName))
+                        {
+                            return;
+                        }
+
+                        fileName = fileName.Split('/')[^1].Split('.')[0];
+
+                        Settings.SettingFunctions.LoadSettings(fileName);
+
+                        fileSelection.Close();
+
+                        // Ask whether the player want to use the loaded settings as default ones
+                        DialogBox dialogBox = new DialogBox();
+                        dialogBox.TitleLabel.Text = "Default settings";
+                        dialogBox.AddText($"Do you want to use '{fileName}.json' as the default settings? (If you agree, the current default settings will be overwritten!)");
+                        dialogBox.OnAgree = () =>
+                        {
+                            Settings.SettingFunctions.SaveSettings();
+
+                            dialogBox.Close();
+                        };
+                        dialogBox.OnDecline = () =>
+                        {
+                            dialogBox.Close();
+                        };
+
+                        FindRootPanel().AddChild(dialogBox);
+
+                        dialogBox.Display();
+                    };
+
+                    fileSelection.Display();
+                });
```

Comment:
I wonder if we could either condense some of this logic or split it up to make it a bit more readable.

Reply:
Already modified in the other PR, but not as much condensed